### PR TITLE
Clean up analysis issues (Resolves #1165)

### LINKED ISSUE #1165: Clean up all Flutter 3.10 and Dart 3 analysis issues
Upgrading to Flutter 3.10 and Dart 3 created a bunch of analysis warnings.

For this ticket, fix all analysis issues.

### PR DESCRIPTION
**Pull request by rutvik110**
Clean up analysis issues (Resolves #1165)

**Description :** 
Fixes analysis issues introduced due to upgrading to Flutter 3.10 and Dart 3

### CODE REVIEW

**Comment by rutvik110** [comment]
hey @matthew-carroll , there are few other non-analytics compile issues introduced in `/website` project. Seems related to class declarations, and non-existing classes/methods issues due to older versions on some packages, seems like around `super_editor` usage?

Do you want me to fix those with this PR?

File: website/lib/homepage/editor_toolbar.dart [hunk 1]
```diff
@@ -288,7 +288,7 @@ class _EditorToolbarState extends State<EditorToolbar> {
 
   /// Takes appropriate action when the toolbar's link button is
   /// pressed.
-  void _onLinkPressed() {
+  void c_onLinkPressed() {
```
Comment (matthew-carroll) [inline website/lib/homepage/editor_toolbar.dart:291]: Why was a "c" added to the name?

File: website/lib/homepage/footer.dart [hunk 2]
```diff
@@ -144,7 +144,7 @@ class _Link extends StatelessWidget {
       cursor: SystemMouseCursors.click,
       child: GestureDetector(
         behavior: HitTestBehavior.opaque,
-        onTap: () => launch(url),
+        onTap: () => launchUrl(Uri.parse(url)),
```
Comment (matthew-carroll) [inline website/lib/homepage/footer.dart:147]: Was `launch()` deprecated?

**Review by matthew-carroll — COMMENTED** [review]
Generally LGTM. I left a few comments/questions.

**Comment by matthew-carroll** [comment]
> hey @matthew-carroll , there are few other non-analytics compile issues introduced in `/website` project. Seems related to class declarations, and non-existing classes/methods issues due to changes in `super_editor` usage?
> 
> Do you want me to fix those with this PR?

Sure, feel free to fix those here.

File: super_editor/example/lib/demos/demo_animated_task_height.dart [hunk 3]
```diff
@@ -98,8 +98,9 @@ class AnimatedTaskComponentBuilder implements ComponentBuilder {
 class _AnimatedTaskComponent extends StatefulWidget {
   const _AnimatedTaskComponent({
     Key? key,
-    required this.viewModel,
+    // ignore: unused_element
     this.showDebugPaint = false,
+    required this.viewModel,
```
[the remark this answers is not in the document]
Comment (rutvik110) [reply super_editor/example/lib/demos/demo_animated_task_height.dart:103]: That must've been a mistake on my side. Will update it.

File: super_editor/example/lib/demos/example_editor/example_editor.dart [hunk 4]
```diff
@@ -30,7 +30,7 @@ class _ExampleEditorState extends State<ExampleEditor> {
 
   final _darkBackground = const Color(0xFF222222);
   final _lightBackground = Colors.white;
-  ValueNotifier<Brightness> _brightness = ValueNotifier<Brightness>(Brightness.light);
+  final ValueNotifier<Brightness> _brightness = ValueNotifier<Brightness>(Brightness.light);
```
[the remark this answers is not in the document]
Comment (rutvik110) [reply super_editor/example/lib/demos/example_editor/example_editor.dart:33]: Will do.

File: website/lib/homepage/editor_toolbar.dart (hunk 1 above)
Comment (rutvik110) [reply website/lib/homepage/editor_toolbar.dart:291]: that must've been added accidentally. will update it.

File: website/lib/homepage/footer.dart (hunk 2 above)
Comment (rutvik110) [reply website/lib/homepage/footer.dart:147]: yah, I updated it in places it was used with appropriate launcher.

**Comment by rutvik110** [comment]
> > hey @matthew-carroll , there are few other non-analytics compile issues introduced in `/website` project. Seems related to class declarations, and non-existing classes/methods issues due to changes in `super_editor` usage?
> > Do you want me to fix those with this PR?
> 
> Sure, feel free to fix those here.

Cool!

**Comment by rutvik110** [comment]
Just fixed the remaining issues and the website project now compiles and builds fine.

Btw, minor issue on editor toolbar on the website, the dropdowns won't appear. Seems like some sort of overlay issue. Should we open up new issue to tackle this?

https://github.com/superlistapp/super_editor/assets/65209850/ceb913e2-9ac1-42ad-bc44-6b4b33de4112

**Review by matthew-carroll — APPROVED** [review]
LGTM. I'll go ahead and merge this. For future reference, notice that "squash and merge" to `main`. This gives us a single commit for the work in this PR.

Once this is merged to `main`, in a few minutes, please branch off of `stable`, cherry pick this commit from `main`, and then put up a PR against `stable`.

When we merge that one, we'll use GitHub's "rebase" option so that we retain the same commit ID from `main`.

**Comment by matthew-carroll** [comment]
@rutvik110 WRT the toolbar - can you check if we already have a ticket for that? I think I've seen it somewhere.

**Comment by rutvik110** [comment]
Sure!

**Comment by rutvik110** [comment]
#1044  and #1139, this two are referring to the same issue.

**Pull request merged**